### DISCUSSION TITLE
Exclude cache for dynamic data

### DISCUSSION THREAD

**Discussion by angrybrad** [craftcms/cms#8112] [Ideas]
> Created by: Johannes Lamers (johanneslamers@gmail.com) on 2015/03/10 17:03:25 +0000
> Votes at time of UserVoice import: 49

-----------------------------------

Exclude parts of a cached item for dynamic user info.

In CE Cache for EE you have:

{exp:ce_cache:escape}
<p>{if logged_in}Hello {username}{if:else}Please log in{/if}</p>
{/exp:ce_cache:escape}

**Comment by angrybrad** [comment]
> Posted by Jeremy Daalder (jd@imagescience.com.au) on 2015/08/17 22:59:59 +0000
-----------------------------------

I honestly think, given how big a part of things Commerce now is, that this is something that should be in core - the abaility to escape reguions from being cached.

Commerce by definition means stacks of form submissions, which means CSRF is desirable, but of course so is performance and thus caching, so....ya kinda need it really.

**Comment by angrybrad** [comment]
> Posted by Simon Leadbetter (simon@thoughtfulweb.com) on 2015/04/18 14:12:43 +0000
-----------------------------------

I’d love to see this implemented. I’ve struggled applying cache to a site that flags member only content in listings etc.

**Comment by angrybrad** [comment]
> Posted by Kristen Grote (kristen@madewithloveco.com) on 2015/12/01 23:48:33 +0000
-----------------------------------

I could REEEAAAAALY use this. It&#39;s very hard to keep my caching clean and organized without it.

**Comment by angrybrad** [comment]
> Posted by Joel Bradbury (joel@squarebit.co.uk) on 2015/03/18 14:30:51 +0000
-----------------------------------

As Johannes mentioned on SE, a use case would be : https://craftcms.stackexchange.com/questions/8966/cache-entry-but-escape-dynamic-data

**Comment by arifje** [comment]
Any chance this will be implemented?

**Comment by brandonkelly** [comment]
There’s not really a clean way to do it that won’t lead to a lot of confusion, so not sure. You can do something like this though.

```twig
{% set cachedHtml %}
  {% cache %}
    <!-- ... -->
    GREETING_PLACEHOLDER
    <!-- ... -->
  {% endcache %}
{% endset %}

{% set greeting %}
  {% if currentUser %}
    <p>Hello, {{ currentUser.friendlyName }}</p>
  {% endif %}
{% endset %}

{{ cachedHtml|replace('GREETING_PLACEHOLDER', greeting)|raw }}
```
